x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/ApiKey.java

```diff
@@ -221,7 +240,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     public XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.field("id", id).field("name", name).field("creation", creation.toEpochMilli());
+        builder.field("id", id).field("name", name);
+        if (TcpTransport.isUntrustedRemoteClusterEnabled()) {
+            builder.field("type", type.value());
+        }
```

Comment:
This PR has some unfortunate complexity in tests due to this conditional serialization logic with featureFlag checking. A simple solution could be adding `assumeTrue(featureFlag)` for all those tests. But it felt disabling a bit too much. So I took a more finer controlled approach. 

Luckily Doc tests do not actually assert Get/Query responses and Yaml tests only assert selective fields. So they need no change in this PR.